packages/flutter_test/lib/src/test_text_input.dart

```diff
@@ -116,83 +146,65 @@ class TestTextInput {
     }
   }
 
-  /// Whether the onscreen keyboard is visible to the user.
-  bool get isVisible {
+  /// Simulates the user hiding the onscreen keyboard.
+  ///
+  /// This does nothing but set the internal flag.
+  void hide() {
     assert(isRegistered);
-    return _isVisible;
+    _isVisible = false;
   }
-  bool _isVisible = false;
 
-  /// Simulates the user changing the [TextEditingValue] to the given value.
-  void updateEditingValue(TextEditingValue value) {
-    assert(isRegistered);
-    // Not using the `expect` function because in the case of a FlutterDriver
-    // test this code does not run in a package:test test zone.
-    if (_client == 0)
-      throw TestFailure('Tried to use TestTextInput with no keyboard attached. You must use WidgetTester.showKeyboard() first.');
-    _binaryMessenger.handlePlatformMessage(
-      SystemChannels.textInput.name,
-      SystemChannels.textInput.codec.encodeMethodCall(
-        MethodCall(
-          'TextInputClient.updateEditingState',
-          <dynamic>[_client, value.toJSON()],
-        ),
-      ),
-      (ByteData? data) { /* response from framework is discarded */ },
-    );
+  /// Simulates the user typing the given text.
+  ///
+  /// This can be called even if the [TestTextInput] has not been [register]ed.
+  ///
+  /// If this is used to inject text when there is a real IME connection, for
+  /// example when using the [integration_test] library, there is a risk that
+  /// the real IME will become confused as to the current state of input.
+  void enterText(String text) {
+    updateEditingValue(TextEditingValue(
+      text: text,
```

Comment:
Should the selection be set to the end of the text?

Reply:
@LongCatIsLooong is doing that in https://github.com/flutter/flutter/pull/79506 but it got reverted due to google3 breakage.

Reply:
Ah I thought that sounded familiar 👍